### PR TITLE
Feature/create request use case

### PR DESCRIPTION
**Pull request by armaan-n**
TLDR; Implemented functionality and the view necessary for the user to create a service request. Test suite should be extended, but is appropriate for pull request. I also added some empty interface adapters that need to be implemented.

# Requirements : 

The user should be able to create a request by specifying a destination, urgency level and service. The user should automatically be matched with a doctor, create the necessary request, mark the doctor as busy, add the request to the patient's list. In the case that a doctor cannot be matched (i.e. all doctors are busy), remain on the same screen and display an error message. The requested doctor should specialize in the requested service if possible, but if they are an hour or more away, match the patient with the closest doctor.

# Implementation Details :

Price, eta and distance parameters are determined via the `apiAccessInterface`, doctor matching is performed with the `DoctorMatcher` (see entities package), the available doctors are determined via the `doctorDataAccessInterface`, and patient data is manipulated via the `patientDataAccessInterface`. Concrete implementations of the api/data access interfaces should be created for final app. All other data needed to create the request is provided via the input data object.

The user and doctor data are saved via the `patientDataAccessInterface` and `doctorDataAccessInterface`.

The fail presenter should be called whenever there are no available doctors, or the user has inputted an invalid location. The success presenter should be called otherwise, passing the request and patient as output data.

# Testing :

A non-comprehensive test suite has been created for features relating to the use case. To decouple the tests from implementations of the access objects, artificial data/api access objects are implemented as private classes within the test classes.

The doctor matcher should be tested for :

- When there is an available specialized doctor (alongside non-specialized doctors) and they are less than 60 minutes away (this should result in matching with the specialized doctor). 
- When there is no specialized doctor (in this case, match with the closest doctor)
- When there is a specialized doctor but they are too far away (in this case, match with the closest doctor)
- When there are no available doctors (raise exception)

The use case interactor should be tested for :

- When the user successfully matches with a doctor (call success presenter and ensure valid output data)
- When the user can't match with a doctor, either because the location was invalid, or there were no available doctors (call fail presenter)

# Notes :

The test suit should be extended in the future, it's not sufficient for our final submission.  E.x. add tests for when there are multiple specialized doctors.

I've also implemented a return home use case, but it is minimal in scope, and is intended to be a helper that can be used by many use cases. 

I've also added many empty interface adapters related to use cases this feature interacts with. These should guide the development of our app, and aid with the consistency of file naming.

### CODE REVIEW

**Review by michellengnx — APPROVED** [review]
image

Umm... LGTM, actually. Great job on documenting! Too long, but I did read 💯 

Agree on the test suit. Also missing some of unit tests in entities. We will need to add plugin to check for test coverage before final submission (see [#26](https://github.com/michellengnx/csc207-project/issues/26)).
